src/code/z_bgcheck.c

```diff
@@ -4112,10 +4112,13 @@ s32 SurfaceType_IsIgnoredByProjectiles(CollisionContext* colCtx, CollisionPoly*
 }
 
 /**
- * CollisionPoly is conveyor enabled
- * Returns true if `poly` is a conveyor surface, else false
+ * CollisionPoly is a floor conveyor
+ * 
+ * A conveyor surface is enabled with non-zero speed. When enabled, the conveyor will exhibit two types of behaviour depending on the return value:
```

Comment:
I don't think this is going to work out well with the formatter. The lines would probably need to be broken up further.

Also I'm not sure it's right to have documentation here about what the function is used for in player. The function could very well be used for other purposes in a modding scenario for instance. Imo the comment should just describe what the function itself does, which is to return whether the surface is a floor conveyor and not a water one.

Reply:
Well at that point you could also argue it shouldn't be called "conveyor" either since that name is derived from the effect this has on player, for example it could be named as "get boolean number 4" and then it could very well be used for other purposes in a modding scenario for instance.

I'm not sure where you'd draw the line

Reply:
I'll fix the formatting!

Just to add what Dragorn said, even the fact that this is characterized by a "floor" is based on what the function is used for in Player, in the exact same block that characterizes is not a water one. Describing only what the function does is to extract a specific property from the surface, `poly->flags_vIB & 0x2000`, and nothing more.

Actually naming and documenting what this property does will depend on how it's used in code, which in this case it's player. Of course one could mod a property to have different behaviour. But that shouldn't prevent us from documenting surface properties for what they actually do in game. Does that make sense?

Perhaps there's somewhere else this description can go?